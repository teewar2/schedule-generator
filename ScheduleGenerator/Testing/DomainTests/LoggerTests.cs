using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using Domain;
using NUnit.Framework;
using static Domain.DomainExtensions;
using static Testing.ObjectMother;

namespace Testing.DomainTests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class LoggerTests
    {
        [Test]
        public void CheckLogger()
        {
            var estimator = GetDefaultCombinedEstimator();
            var schedule = new Schedule(FullMondayRequisition, ClassroomsRequisitions);

            while (true)
            {
                var meetingToAdd = schedule.GetMeetingsToAdd().FirstOrDefault();
                if (meetingToAdd == null) break;
                schedule.AddMeeting(meetingToAdd);
            }

            using var logger = new Infrastructure.Logger("TestLogger");
            estimator.Estimate(schedule, logger);
            var loggerString = logger.ToString();
            Approvals.Verify(loggerString);
        }
    }
}