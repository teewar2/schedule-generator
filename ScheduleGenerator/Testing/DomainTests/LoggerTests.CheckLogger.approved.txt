StudentsSpacesEstimator:
	ФИИТ-101 Part1 has 1 spaces on Even Monday
	ФИИТ-101 Part2 has 1 spaces on Even Monday
	Total: -0,020833333333333332 (BasicScore: -0,020833333333333332, Weight: 1)
MeetingsPerDayEstimator:
	ФИИТ-101 Part1 has bad Odd Monday with 6 meetings
	ФИИТ-101 Part1 has bad Even Monday with 5 meetings
	ФИИТ-101 Part2 has bad Odd Monday with 6 meetings
	ФИИТ-101 Part2 has bad Even Monday with 5 meetings
	Total: -0,16666666666666666 (BasicScore: -0,16666666666666666, Weight: 1)
TeacherSpacesEstimator:
	Филатова has 1 spaces on Even Monday
	Total: -0,006944444444444444 (BasicScore: -0,006944444444444444, Weight: 1)
Total: -0,19444444444444445 (BasicScore: -0,19444444444444445, Weight: 1)